api/server/router/swarm/cluster_routes.go

```diff
@@ -218,13 +218,23 @@ func (sr *swarmRouter) createService(ctx context.Context, w http.ResponseWriter,
 				// Sysctls for docker swarm services weren't supported before
 				// API version 1.40
 				service.TaskTemplate.ContainerSpec.Sysctls = nil
+
+				if service.TaskTemplate.ContainerSpec.Privileges != nil && service.TaskTemplate.ContainerSpec.Privileges.CredentialSpec != nil {
+					// Support for setting credential-spec through configs was added in API 1.40
+					service.TaskTemplate.ContainerSpec.Privileges.CredentialSpec.Config = ""
+				}
+				for _, config := range service.TaskTemplate.ContainerSpec.Configs {
+					// support for the Runtime target was added in API 1.40
+					config.Runtime = nil
```

Comment:
A Config with no Target is invalid. But I believe that having a Runtime-type Config is important because it affirms the intended purpose of the config.

A Config with both a Runtime and a File should be invalid, but I believe that the current implementation would just favor Runtime and omit the File portion entirely. I'm unsure where the validation check should be.

Runtime isn't the best name but that's what I called it in Swarmkit and that's what we merged, so I'm committing to it.